posts/creating-your-first-truffle-project.md

```diff
@@ -0,0 +1,634 @@
+            throw new Error('The environment variable "MNEMONIC" must be 12 words (space delineated)');
+        }
+        
+        ...
+        
+        modules.export = {
+            networks: {
+                ...
+                kovan: {
+                    provider: () =>
+                        new HDWalletProvider(mnemonic, `https://kovan.infura.io/v3/${infuraProjectId}`),
+                    network_id: 42, // Kovan Id
+                    gas: 3000000,
+                    gasPrice: 100000000000
+                },
+                rinkeby: {
+                    provider: () =>
+                        new HDWalletProvider(mnemonic, `https://rinkeby.infura.io/v3/${infuraProjectId}`),
+                    network_id: 4, // Rinkeby Id
+                    gas: 3000000,
+                    gasPrice: 100000000000
+                },
+                ropsten: {
+                    provider: () =>
+                        new HDWalletProvider(mnemonic, `https://ropsten.infura.io/v3/${infuraProjectId}`),
+                    network_id: 3, // Ropsten Id
+                    gas: 3000000,
+                    gasPrice: 100000000000
+                },
+                live: {
+                    provider: () =>
+                        new HDWalletProvider(mnemonic, `https://mainnet.infura.io/v3/${infuraProjectId}`),
+                    network_id: 1, // Mainnet Id
+                    gas: 4000000,
+                    gasPrice: 100000000000
+                }
+            },
+            ...
+        }
+    We have to use the mnemonic here, because when we start sending transactions to the main networks, we have to use existing addresses with balances. 
+    
+3. Now we need to install [HDWalletProvider](https://github.com/trufflesuite/truffle-hdwallet-provider) so we can turn mnemonics into our addresses and private keys.
+
+        npm install truffle-hdwallet-provider --save-dev
+
+4. Try running one or all of the forking scripts:
+        
+        npm run start:kovan
+
+This command will start our local ganache (which forked from kovan), and then will migrate our contracts to it. Our local client will keep running, and we can view the contracts which have been successfully deployed.
+
+Run `node_modules/.bin/truffle networks`: ![Truffle network (pic missing)](https://raw.githubusercontent.com/tylerjohnhaden/blog-usa/master/images/2019/01/truffle_networks.png)
+
+You can send function calls as transactions to this address on our local client. 
+
+# Adding linting to your project
+
+When working with contract code, you should always follow [best practices](https://solidity.readthedocs.io/en/v0.5.2/security-considerations.html#security-considerations). Not because your technical lead is particularly stubborn, but because this code will be handling your client's money. You may assume that your solidity code will be audited for security and optimization. Linting is always the first step in this process.
```

Comment:
```suggestion
When working with contract code, you should always follow [best practices](https://solidity.readthedocs.io/en/v0.5.2/security-considerations.html#security-considerations). Not because your technical lead is particularly stubborn, but because this code will be handling your client's money. You may assume that your Solidity code will be audited for security and optimization. Linting is always the first step in this process.
```